git-hub

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+import subprocess
+import github
+import yaml
+import click
+import sys
+
+
+def pr(num):
+    # runs git and gets user and repo of current folder.
+    process = subprocess.Popen("git remote -v", stdout=subprocess.PIPE)
+    remotes = str(process.stdout.read())
+    url = remotes.split(" ", 1)[0]  # gets the fetch url
+    url = url.split("\\t", 1)[1]  # takes out extra info
+    arguments = url.split(".com:")[1]  # gets just the username/repo.git
+    arguments = arguments[:len(arguments) - 4]  # takes out ".git"
+    username, repo = arguments.split("/")
+
+    # gets token from config folder.
+    try:
+        with open(".config/git-hub.yaml") as stream:
+            yaml_file = str(yaml.load(stream))
+            token = yaml_file.split("=")[1].strip()
+    except (IndexError, FileNotFoundError) as e:
+        print("Needs an authentification token in ~/.config/git-hub.yaml")
+        print("file simply says:")
+        print("token = abc123")
+        print("""Visit https://help.github.com/articles/
+            creating-a-personal-access-token-for-the-command-line/""")
+        print("to learn how to generate a token.")
+        sys.exit(1)
+
+    # gets pr and runs command.
+    try:
+        g = github.Github(token)
+        pr = g.get_user(username).get_repo(repo).get_pull(num)
+        label = pr.head.label
+        other_user, branch = label.split(":")
+        remote_name = "\\n" + other_user + "\\t"
```

Comment:
What is happening here?

Reply:
The label of a pr on Github's API is represented as "user: branch" and I split it by ":". to get the user and branch. I already had the remote name from before, but I added the "/n" and "/t" to so that in the next line when I check if the remote_name is in remotes, it won't detect that a remote is in remotes if it is a smaller string(e.g. suppose you had a remote named "ttran428" and now you pulled from a pr whose username is "tran42". If I don't have the "/n" and "/t" it will detect that "tran42" has already been added as a remote while in reality, it hasn't since it detected the substring of "ttran428".